web_src/js/index.js

```diff
@@ -357,214 +437,221 @@ function initSimpleMDEImagePaste(simplemde, files) {
+  selectItem('.select-milestone', '#milestone_id');
+  selectItem('.select-assignee', '#assignee_id');
+}
+
 
-    $menu.find('.item:not(.no-select)').on('click', function () {
-      $(this).parent().find('.item').each(function () {
-        $(this).removeClass('selected active');
-      });
+function selectItem(select_id, input_id) {
+  const $menu = $(`${select_id} .menu`);
+  const $list = $(`.ui${select_id}.list`);
+  const hasUpdateAction = $menu.data('action') === 'update';
 
-      $(this).addClass('selected active');
-      if (hasUpdateAction) {
-        updateIssuesMeta(
-          $menu.data('update-url'),
-          '',
-          $menu.data('issue-id'),
-          $(this).data('id'),
-          $(this).data('is-checked')
-        ).then(reload);
-      }
-      switch (input_id) {
-        case '#milestone_id':
-          $list.find('.selected').html(`<a class="item" href=${$(this).data('href')}>${
-            htmlEscape($(this).text())}</a>`);
-          break;
-        case '#project_id':
-          $list.find('.selected').html(`<a class="item" href=${$(this).data('href')}>${
-            htmlEscape($(this).text())}</a>`);
-          break;
-        case '#assignee_id':
-          $list.find('.selected').html(`<a class="item" href=${$(this).data('href')}>` +
-                        `<img class="ui avatar image" src=${$(this).data('avatar')}>${
-                          htmlEscape($(this).text())}</a>`);
-      }
-      $(`.ui${select_id}.list .no-select`).addClass('hide');
-      $(input_id).val($(this).data('id'));
+  $menu.find('.item:not(.no-select)').on('click', function () {
+    $(this).parent().find('.item').each(function () {
+      $(this).removeClass('selected active');
     });
-    $menu.find('.no-select.item').on('click', function () {
-      $(this).parent().find('.item:not(.no-select)').each(function () {
-        $(this).removeClass('selected active');
-      });
-
-      if (hasUpdateAction) {
-        updateIssuesMeta(
-          $menu.data('update-url'),
-          '',
-          $menu.data('issue-id'),
-          $(this).data('id'),
-          $(this).data('is-checked')
-        ).then(reload);
-      }
 
-      $list.find('.selected').html('');
-      $list.find('.no-select').removeClass('hide');
-      $(input_id).val('');
+    $(this).addClass('selected active');
+    if (hasUpdateAction) {
+      updateIssuesMeta(
+        $menu.data('update-url'),
+        '',
+        $menu.data('issue-id'),
+        $(this).data('id'),
+        $(this).data('is-checked')
+      ).then(reload);
+    }
+    switch (input_id) {
+      case '#milestone_id':
+        $list.find('.selected').html(`<a class="item" href=${$(this).data('href')}>${
+          htmlEscape($(this).text())}</a>`);
+        break;
+      case '#project_id':
+        $list.find('.selected').html(`<a class="item" href=${$(this).data('href')}>${
+          htmlEscape($(this).text())}</a>`);
+        break;
```

Comment:
Can just fallthrough that case, e.g. 

```
case '#milestone_id':
case '#project_id':

Reply:
Ok but this fails with eslint, had to add // falls through comment https://eslint.org/docs/rules/no-fallthrough

Reply:
Right, fallthrough can be treachorous. I'd suggest rewriting to a `if-else-if` chain.